xbmc/windowing/osx/WinSystemIOS.mm

```diff
@@ -266,69 +269,66 @@ - (void) runDisplayLink;
   //and push to the resolution info vector
-  FillInVideoModes();
+  FillInVideoModes(screenIdx);
 }
 
-void CWinSystemIOS::FillInVideoModes()
+void CWinSystemIOS::FillInVideoModes(int screenIdx)
 {
   // Add full screen settings for additional monitors
-  int numDisplays = GetNumScreens();
-
-  for (int disp = 0; disp < numDisplays; disp++)
+  RESOLUTION_INFO res;
+  int w, h;
+  // atm we don't get refreshrate info from iOS
+  // but this may change in the future. In that case
+  // we will adapt this code for filling some
+  // useful info into this local var :)
+  double refreshrate = 0.0;
+  //screen 0 is mainscreen - 1 has to be the external one...
+  UIScreen *aScreen = [[UIScreen screens]objectAtIndex:screenIdx];
+  //found external screen
+  for ( UIScreenMode *mode in [aScreen availableModes] )
   {
-    RESOLUTION_INFO res;
-    int w, h;
-    // atm we don't get refreshrate info from iOS
-    // but this may change in the future. In that case
-    // we will adapt this code for filling some
-    // useful info into this local var :)
-    double refreshrate = 0.0;
-    //screen 0 is mainscreen - 1 has to be the external one...
-    UIScreen *aScreen = [[UIScreen screens]objectAtIndex:disp];
-    //found external screen
-    for ( UIScreenMode *mode in [aScreen availableModes] )
+    w = mode.size.width;
+    h = mode.size.height;
+    
+    if (screenIdx == 0)
     {
-      w = mode.size.width;
-      h = mode.size.height;
-      UpdateDesktopResolution(res, disp, w, h, refreshrate);
-      CLog::Log(LOGNOTICE, "Found possible resolution for display %d with %d x %d\n", disp, w, h);
-
-      //overwrite the mode str because  UpdateDesktopResolution adds a
-      //"Full Screen". Since the current resolution is there twice
-      //this would lead to 2 identical resolution entrys in the guisettings.xml.
-      //That would cause problems with saving screen overscan calibration
-      //because the wrong entry is picked on load.
-      //So we just use UpdateDesktopResolutions for the current DESKTOP_RESOLUTIONS
-      //in UpdateResolutions. And on all other resolutions make a unique
-      //mode str by doing it without appending "Full Screen".
-      //this is what linux does - though it feels that there shouldn't be
-      //the same resolution twice... - thats why i add a FIXME here.
-      res.strMode = StringUtils::Format("%dx%d @ %.2f", w, h, refreshrate);
-      CServiceBroker::GetWinSystem()->GetGfxContext().ResetOverscan(res);
-      CDisplaySettings::GetInstance().AddResolutionInfo(res);
+      res.strOutput = CONST_TOUCHSCREEN;
     }
+    else
+    {
+      res.strOutput = CONST_EXTERNAL;
+    }
+    
+    UpdateDesktopResolution(res, 0, w, h, refreshrate);
+    CLog::Log(LOGNOTICE, "Found possible resolution for display %d with %d x %d\n", screenIdx, w, h);
+
+    //overwrite the mode str because  UpdateDesktopResolution adds a
+    //"Full Screen". Since the current resolution is there twice
```

Comment:
why don't you drop this. not needed anymore

Reply:
Because it was still there on winsystemosx aswell iirc - will drop :)